app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoFragment.kt

```diff
@@ -19,74 +21,52 @@ import android.widget.ImageView
 import android.widget.TextView
 import kotlinx.android.synthetic.main.fragment_pocket_video.*
 import kotlinx.android.synthetic.main.fragment_pocket_video.view.*
-import kotlinx.coroutines.experimental.CoroutineStart
-import kotlinx.coroutines.experimental.Job
-import kotlinx.coroutines.experimental.android.UI
-import kotlinx.coroutines.experimental.launch
 import mozilla.components.browser.session.Session
 import org.mozilla.tv.firefox.R
 import org.mozilla.tv.firefox.ScreenController
+import org.mozilla.tv.firefox.ViewModelFactory
 import org.mozilla.tv.firefox.ext.forceExhaustive
-import org.mozilla.tv.firefox.ext.isNotCompleted
 import org.mozilla.tv.firefox.ext.resetAfter
 import org.mozilla.tv.firefox.ext.serviceLocator
 import org.mozilla.tv.firefox.ext.updateLayoutParams
 import org.mozilla.tv.firefox.telemetry.TelemetryIntegration
 import org.mozilla.tv.firefox.utils.FormattedDomain
 import org.mozilla.tv.firefox.utils.PicassoWrapper
+import org.mozilla.tv.firefox.utils.ServiceLocator
 import java.net.URI
 import java.net.URISyntaxException
 
 /** A feed of Pocket videos. */
 class PocketVideoFragment : Fragment() {
 
-    private lateinit var deferredVideos: PocketVideosDeferred
-    private var uiLifecycleCancelJob: Job? = null
+    private lateinit var viewModel: PocketViewModel
+    private lateinit var serviceLocator: ServiceLocator
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        // The video feed is typically provided by the creator of this fragment such that the videos
-        // on that view (the home screen) match the ones in this feed. However, if this fragment is
-        // being reconstructed, the data will no longer be in memory so we get retrieve the data again
-        // here. Typically, the data would be provided with `Fragment.getArguments`, which would be
-        // provided to reconstructed fragment but we intentionally don't do that: if the feed is
-        // being reconstructed, the user probably cares more about an up-to-date feed rather than
-        // consistency with the home screen they saw a while ago.
-        if (!this::deferredVideos.isInitialized) {
-            deferredVideos = context!!.serviceLocator.pocket.getRecommendedVideos()
-        }
+        serviceLocator = context!!.serviceLocator
+        val factory = ViewModelFactory(serviceLocator)
+        viewModel = ViewModelProviders.of(this, factory).get(PocketViewModel::class.java)
```

Comment:
Why not do this in `onCreateView` and avoid separating this logic?

In fact, this might break because the viewModel's lifecycle owner is viewLifecycleOwner (I'm not sure if a ViewModel can be re-used at that point).